lib/meadow/pipeline/actions/generate_file_set_digests.ex

```diff
@@ -25,15 +25,21 @@ defmodule Meadow.Pipeline.Actions.GenerateFileSetDigests do
     try do
       Logger.info("Generating digests for #{file_set.id}")
 
-      hashes = generate_hashes(file_set.metadata.location)
-      Logger.info("Hash for #{file_set.id}: #{hashes.sha256}")
-
-      {:ok, _} =
-        file_set
-        |> FileSets.update_file_set(%{metadata: %{digests: hashes}})
-
-      ActionStates.set_state!(file_set, __MODULE__, "ok")
-      :ok
+      case generate_hashes(file_set.metadata.location) do
+        {:ok,
+         %{"sha256" => <<sha256::binary-size(64)>>, "sha1" => <<_sha1::binary-size(40)>>} = hashes} ->
+          Logger.info("Hash for #{file_set.id}: #{sha256}")
+          {:ok, _} = FileSets.update_file_set(file_set, %{metadata: %{digests: hashes}})
+          ActionStates.set_state!(file_set, __MODULE__, "ok")
+          :ok
+
+        {:ok, unexpected_result} ->
+          ActionStates.set_state!(file_set, __MODULE__, "error", unexpected_result)
```

Comment:
Does the `unexpected_result` need to be converted to a string or something in order to be stored? I don't think we know what it is?

Reply:
It ends up looking like:
```
action_states: [
    %Meadow.Data.Schemas.ActionState{
      __meta__: #Ecto.Schema.Metadata<:loaded, "action_states">,
      action: "Meadow.Pipeline.Actions.GenerateFileSetDigests",
      id: "9d204ba7-d897-416a-b739-6b9ed8f9d76e",
      inserted_at: ~U[2021-02-04 20:10:57.965955Z],
      notes: "%{sha1: nil, sha256: nil}",
      object_id: "650b2a56-5aac-4fc7-891f-a306081a9af1",
      object_type: "Meadow.Data.Schemas.FileSet",
      outcome: "error",
      updated_at: ~U[2021-02-04 20:10:58.543300Z]
    }
  ]
  ```

Reply:
Yeah, I think we're fine on this one since MBK's PR got merged.